src/uilib/colorhelper.h

```diff
@@ -92,7 +92,22 @@ static SDL_Surface *makeGreyscale(SDL_Surface *surf, bool darken)
         SDL_SetSurfacePalette(surf, pal);
         SDL_FreePalette(pal); // FIXME: is this required / allowed?
     }
-    else {
+    else if (surf->format->BitsPerPixel == 24) {
+        uint8_t *data = (uint8_t*)surf->pixels;
+        auto fmt = surf->format;
+        for (int y=0; y<surf->h; y++) {
+            for (int x=0; x<surf->w; x++) {
+                uint8_t* ppixel = (uint8_t*)(data+y*surf->pitch+x*fmt->BytesPerPixel);
+                uint8_t w = makeGreyscale(ppixel[0], ppixel[1], ppixel[2], darken);
```

Comment:
This ignores byte order, so RGB source looks different than BGR source because correct greyscale uses illumination, which has weighting on colors (grey(R,G,B) != grey(B,G,R)). I will try to come up with something, unless you are faster :D

This is what SDL provides. it can't be inlined however, so it ends up being really slow for our needs:
https://github.com/libsdl-org/SDL/blob/ad6bc521bd2748a5f2581edf32b4732a1af7ab4c/src/video/SDL_pixels.c#L882